File: UmarAppApi/Models/Purchases/Purchase.cs
using UmarAppApi.Models.Products;

namespace UmarAppApi.Models.Purchases
{
    public class Purchase
    {
        public Guid Id { get; set; }
        public Guid ProductId { get; set; } 
        public int Quantity { get; set; }
        public decimal Price { get; set; }
        public decimal TotalAmount { get; set; }
        public DateTime PurchaseDate { get; set; }
    }
}

File: UmarApp-UI/src/app/core/models/Purchase.ts
export interface Purchase {
  id: string;
  productId: string;
  quantity: number;
  price: number;
  totalAmount: number;
  purchaseDate?: string;
}

export interface PurchaseDto {
  productId: string;
  quantity: number;
  price: number;
  totalAmount: number;
  purchaseDate: string;
}

export interface PurchaseOutputDto {
  id: string;
  productId: string;
  productName: string;
  quantity: number;
  price: number;
  totalAmount: number;
  purchaseDate?: string;
}
